andela/ah-scorpion
# #159053980 Feature JWT Authentication

**What does this PR do?**
Implements a simple version of authenticating users using JSON Web Tokens.

**Description of Task to be completed?**
- Implement a method for generating a JSON Web Token when a user registers or logs in.
- Implement a method for authenticating tokens received from requests to protected endpoints.

**How should this be manually tested?**
- Clone the repository from [here](https://github.com/andela/ah-scorpion/tree/ft-JWT-Auth-%23159053980) and check out the branch `ft-JWT-Auth-#159053980`
- Activate your virtual environment and use the `.env-sample` to set the environment variables.
- Use postman to test these endpoints.
  `POST api/users` for registering
  `POST api/users/login` for logging in
When a user successfully registers or logs in, a JWT should be sent in the response.

- To test whether the token is valid, copy and paste it into the `Authorization` header and prepend it with the keyword `'Bearer'` then access this endpoint:
  `GET api/user` If successful, the response should be the user's details, i.e, the username and email.

What are the relevant pivotal tracker stories?
[#159053980](https://www.pivotaltracker.com/story/show/159053980)
